src/Microsoft.TestPlatform.CrossPlatEngine/Client/CriteriaTransform.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client
+{
+    using System.Collections.Generic;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Host;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
+    using System.Linq;
+
+    internal static class CriteriaTransform
```

Comment:
Suggest to keep the responsibility singular. `Criteria` is a broad term, suggest `PackageTransform`, it will collate all Package related functionality.

Good to see this as a static class with static method 👍

Reply:
moved to extension methods for Discovery and RunCriteria